blosc/blosc.c

```diff
@@ -2117,14 +2118,42 @@ void blosc_set_splitmode(int mode)
   g_splitmode = mode;
 }
 
+/* Child global context is invalid and pool threads no longer exist post-fork.
+ * Discard the old, inconsistent global context and global context mutex and
+ * mark as uninitialized.  Subsequent calls through `blosc_*` interfaces will
+ * trigger re-init of the global context.
+ *
+ * All pthread interfaces have undefined behavior in child handler in current
+ * posix standards: http://pubs.opengroup.org/onlinepubs/9699919799/
+ */
+void blosc_atfork_child(void) {
+  if (!g_initlib) return;
+
+  global_comp_mutex = NULL;
```

Comment:
I believe we could do `pthread_mutex_init()` here instead, and not use a pointer. But I'm fine either way.

Reply:
Just a note, but a call to `pthread_mutex_init` on a previously-init-ed mutex location without a successful `pthread_mutex_destroy` [results in undefined behavior](http://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread_mutex_init.html). I believe that some form of pointer is required so that the value can be abandoned post-fork.